### DISCUSSION TITLE
[Bug] skywalking agent supports logback tid, but after the log is uploaded on grpc, the UI does not display the traceId.

### DISCUSSION THREAD

**Discussion by lakewatcher** [apache/skywalking#12691] [Q&A] [answered]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

<img width="939" alt="999" src="https://github.com/user-attachments/assets/0b802f9c-79e3-4baa-a192-2b624c2c18ae">

for example ,it no link to traceId
<img width="1018" alt="888" src="https://github.com/user-attachments/assets/b8e30926-0d80-4759-abd0-7ae57659f31c">

but  apisix log has a traceId and can be linked to it's trace ,eg: 

<img width="351" alt="777" src="https://github.com/user-attachments/assets/b1a90acd-5de5-4e38-a01e-4c20289e8559">

### What you expected to happen

skywalking agent supports logback tid, but after the log is uploaded on grpc, the UI does not display the traceId.
i don't know .   

### How to reproduce

add logback grpc-log ,agent and start

1.agent is 9.2 . 
2.logback :
```
    <appender name="sky-log" class="org.apache.skywalking.apm.toolkit.log.logback.v1.x.log.GRPCLogClientAppender">
        <encoder class="ch.qos.logback.core.encoder.LayoutWrappingEncoder">
            <layout class="org.apache.skywalking.apm.toolkit.log.logback.v1.x.mdc.TraceIdMDCPatternLogbackLayout">
                <Pattern>%d{yyyy-MM-dd HH:mm:ss.SSS} [%X{tid}] [%thread] %-5level %logger{36} -%msg%n</Pattern>
            </layout>
        </encoder>
    </appender>

    <appender name="ASYNC_SKY" class="ch.qos.logback.classic.AsyncAppender">
        <discardingThreshold>0</discardingThreshold>
        <queueSize>512</queueSize>
        <appender-ref ref="sky-log"/>
    </appender>

    <root level="info">
        <appender-ref ref="ASYNC_SKY"/>
    </root>
```
-javaagent:D:\agent\skywalking-agent-9.2.0\skywalking-agent.jar
-Dskywalking.agent.service_name=trade-demo-app
-Dskywalking.collector.backend_service=xxx:11800

3.other is default . 

### Anything else

Another question ：The parameter:logging.resolver do not affect  GRPCLogClientAppender ， the code is set to TEXT

https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/configurations/
logging.resolver | Logger resolver: PATTERN or JSON. The default is PATTERN, which uses logging.pattern to print traditional text logs. JSON resolver prints logs in JSON format.

```
LogData.Builder builder = LogData.newBuilder()
                .setTimestamp(event.getTimeMillis())
                .setService(Config.Agent.SERVICE_NAME)
                .setServiceInstance(Config.Agent.INSTANCE_NAME)
                .setTags(logTags.build())
                .setBody(LogDataBody.newBuilder().setType(LogDataBody.ContentCase.TEXT.name())
```

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment, marked as answer]
Because you are using async append, that doesn't work. grpc log reporter has internal async mechanism, you don't need to worry that.

**Reply by lakewatcher** [reply]
it is useful. I tried it, synchronization works.

**Reply by lakewatcher** [reply]
This solved my main problem. By the way, can GRPCLogClientAppender send JSON format?

**Reply by wu-sheng** [reply]
That doesn't make sense. This is a grpc log reporter, it is encoded through protobuf.

**Reply by lakewatcher** [reply]
Got it
